test/topology_custom/test_mv_add_nodes.py

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import asyncio
+import pytest
+import time
+import logging
+
+from cassandra.cluster import ConnectionException, ConsistencyLevel, NoHostAvailable, Session, SimpleStatement  # type: ignore
+
+from test.pylib.manager_client import ManagerClient
+from test.topology.conftest import skip_mode
+
+
+logger = logging.getLogger(__name__)
+
+# This test reproduces issues #17786 and #18709
+# In the test, we create a keyspace with a table and a materialized view.
+# We then start writing to the table, causing the materialized view to be updated.
+# While the writes are in progress, we add nodes to the cluster.
+# The test verifies that the node doesn't crash as a result of the node addition,
+# and that the materialized view is eventually consistent.
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_add_node_during_mv_writes(manager: ManagerClient):
```

Comment:
I think you should add sleep injections to increase the probability of encountering different e_r_ms:
- one sleep in MV update code between obtaining the different e_r_ms
- one sleep in the code which updates e_r_ms in storage_service

(i.e., the same injections that @gleb-cloudius did in his reproducers)

Otherwise this test is very unlikely to reproduce either problem.

You should verify that this test is really a regression test. I.e., run it before and after the fix, and verify that it reproduces both failures before the fix, but reproduces neither after the fix.

In fact I believe we may need two tests actually. How can it reproduce #17786? There is no decommission done in this test... only bootstraps.

Reply:
One test would be enough, if it does both bootstrap and decommission (concurrent to doing MV updates), but you should verify that it reproduces both before the fix (e.g. by commenting out bootstrap, checking that it reproduces the decommission issue, then bringing bootstrap back but commenting out decommission, checking that it reproduces the bootstrap issue)

Reply:
I remember @gleb-cloudius that you **did not** manage to reproduce the decommission case with hundreds of runs until you put the sleep in e_r_m update code.

So we do need at least that sleep to hit the decommission case. Not sure about the bootstrap case -- what was your reproducibility without sleep?

Reply:
> I remember @gleb-cloudius that you **did not** manage to reproduce the decommission case with hundreds of runs until you put the sleep in e_r_m update code.
> 

I do not remember all attempts, but it was definitely not easily reproducible.  Lets run the test 10 time and if it does not fail then we need sleeps like in my examples.